flowmachine/tests/test_dataframe_caching.py

```diff
@@ -0,0 +1,71 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+import pandas as pd
+
+import pytest
+
+from flowmachine.features import EventTableSubset
+
+# TODO: These tests are for the in memory cache of recently retrieved dataframes once flowmachine is no longer used to get dataframes they should be removed
+
+
+def test_turn_off_caching():
+    """
+    *.turn_off_caching() 'forgets' generated dataframe.
+    """
+    sd = EventTableSubset("2016-01-01", "2016-01-02")
+    sd.get_dataframe()
+    sd.turn_off_caching()
+    with pytest.raises(AttributeError):
+        sd._df
+
+
+def test_turn_off_caching_handles_error():
+    """
+    *.turn_off_caching() handles AttributeError.
```

Comment:
Minor note: there is no hint in the test code that an AttributeError is raised anywhere in this code snippet. Might be worth thinking about what this test actually tries to test from the view of the interface that `EventTableSubset` (and also other classes?) present to the outside and either change the docstring to reflect that, or even rewrite the test to only use public-facing properties. Suggestion for the docstring: "*.turn_off_caching() works without `._df` attribute being present"